src/sentry/static/sentry/app/components/search/apiSearch.jsx

```diff
@@ -0,0 +1,149 @@
+import {flatten, debounce} from 'lodash';
+import {withRouter} from 'react-router';
+import PropTypes from 'prop-types';
+import Raven from 'raven-js';
+import React from 'react';
+
+import {Client} from '../../api';
+import withLatestContext from '../../utils/withLatestContext';
+
+class ApiSearch extends React.Component {
+  static propTypes = {
+    query: PropTypes.string,
+    /**
+     * Render function that passes:
+     * `isLoading` - loading state
+     * `allResults` - All results returned from all queries: [searchIndex, model, type]
+     * `results` - Results array filtered by `this.props.query`: [searchIndex, model, type]
+     */
+    children: PropTypes.func.isRequired,
+  };
+
+  constructor(props, ...args) {
+    super(props, ...args);
+    this.state = {
+      loading: false,
+      allResults: null,
+      results: null,
+    };
+
+    this.api = new Client();
+
+    if (props.query) this.doSearch(props.query);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.query !== this.props.query) {
+      this.setState({loading: true});
+      this.doSearch(nextProps.query);
+    }
+  }
+
+  doSearch = debounce(async query => {
+    let {params, organization} = this.props;
+    let orgId = params.orgId || (organization && organization.slug);
+    let urls = ['/organizations/'];
+
+    // Only run these queries when we have an org in context
+    if (orgId) {
+      urls = [
+        ...urls,
+        `/organizations/${orgId}/projects/`,
+        `/organizations/${orgId}/teams/`,
+        `/organizations/${orgId}/members/`,
```

Comment:
probably drop this one

Reply:
oh, wait, they do exist

Reply:
they all point to http://localhost:8000/settings/sentry/member/undefined/ right now though